Nav bar

Cililocwee opened the pull request:
## Type of change
<!-- Choose 1-2 option(s) which best outline the pull request -->
[x] New feature
<br />
[] Bug fix
<br />
[] Documentation
<br />
[] Code refactor/tech debt
<br />
[] Infrastructure


## Description
This implements a responsive navbar that has a mobile view (drop down)


## Ticket link
[Ticket](https://linear.app/cdw/issue/CDW-43/style-the-navbar)


## Part of a back end implementation?
[x] No
<br />
[] [Yes](<!-- Add the link to your back end PR here -->)  


## How to test
<!-- If your ticket is testable outline the test steps
    Example:
        1) Checkout to this branch and run the app
        2) Navigate to /pageName
        3) Click on [something]
        4) [something] should do [action] -->

1. Checkout this branch and run frontend
2. Check nav buttons by navigating between pages
3. Resize window to check repsonsiveness
4. Toggle mobile view in devtools to see dropdown hamburger menu
5. Click hamburger menu to make sure dropdown toggles


## Screenshots
<!-- For UI updates/additions -->
![image](https://github.com/jtabba/CDW-1/assets/12403004/9034ee8b-5d38-4c44-93e8-fa632431818f)
![image](https://github.com/jtabba/CDW-1/assets/12403004/462c528a-a969-4e16-8600-d203c3f20e4a)
![image](https://github.com/jtabba/CDW-1/assets/12403004/3289e31d-0987-4d2d-977f-4c4ae52bbbd4)

Cililocwee committed:
Basic structure

Basic structure of navbar that hides on mobile. Need to implement mobile nav

Cililocwee committed:
Navbar positioning fixed on mobile

Fixed issue with mobile navbar being pushed off screen

Cililocwee committed:
Fixed Navbar to top

Fixing the nav to top included adjust the padding on the main .App and pulling the entire header up to absolute top

Cililocwee committed:
dropdown menu padding

For visibility, added a set width with padding and a dropshadow to the drop down nav menu

Cililocwee committed:
Minor routing fix

Fixed the routing to home page (previous the default route only routed to the UI)

jtabba commented on frontend/src/components/UserInterface.tsx:
```diff
@@ -3,14 +3,14 @@ import { Header } from "./header/Header";
-				<Header />
+  return (
+    <Container margin="auto" maxWidth="90%" padding={"28px 0px"} pt={20}>
+      <Grid templateRows="repeat(3, auto)" gap={1}>
```
Very good

jtabba commented on frontend/src/components/header/Header.tsx:
```diff
@@ -2,23 +2,28 @@ import { Flex } from "@chakra-ui/react";
+      left={0}
+      w={"100dvw"}
+      px={8}
+    >
```
Getting a bit chunky, lets export it into a local `styles.ts` folder

Cililocwee replied:
I'm trying to implement this change now, but a big problem I'm running into is that by styling the component with the style property overrides all the chakra stylings that I'm NOT styling:

When I use style.ts:
![image](https://github.com/jtabba/CDW-1/assets/12403004/676dcc45-78f6-4218-aec4-bfd7cdc806f4)

When I style the component using the chakra props:

![image](https://github.com/jtabba/CDW-1/assets/12403004/8b77d741-8388-4c44-8b49-e88043343f64)

jtabba commented on frontend/src/components/header/NavBar.tsx:
```diff
@@ -1,9 +1,152 @@
-		<Link to={"/mentors"}>Mentors</Link>
-	</>
-);
+"use client";
```
What's this for?

jtabba commented on frontend/src/components/header/NavBar.tsx:
```diff
@@ -1,9 +1,152 @@
+          borderBottom={1}
+          borderStyle={"solid"}
+          borderColor={useColorModeValue("gray.200", "gray.900")}
+          align={"center"}
```
Same with above

jtabba commented on frontend/src/components/header/NavBar.tsx:
```diff
@@ -1,9 +1,152 @@
+  );
+};
+
+const MobileNavItem = ({ label, children, href }: NavItem) => {
```
Use FC

jtabba commented on frontend/src/components/header/NavBar.tsx:
```diff
@@ -1,9 +1,152 @@
+  children?: Array<NavItem>;
+  href?: string;
+}
+
```
Move to top for easier readability with props

jtabba commented on frontend/src/components/header/NavBar.tsx:
```diff
@@ -1,9 +1,152 @@
+        textAlign={"center"}
+        _hover={{
+          textDecoration: "none",
+        }}
```
Move to `styles` file

jtabba commented on frontend/src/components/header/NavBar.tsx:
```diff
@@ -1,9 +1,152 @@
+            _hover={{
+              textDecoration: "none",
+              color: linkHoverColor,
+            }}
```
`styles`

jtabba commented on frontend/src/components/header/NavBar.tsx:
```diff
@@ -1,9 +1,152 @@
+    label: "Mentors",
+    href: "mentors",
+  },
+];
```
I'd suggest moving to own file with the component that uses it

jtabba reviewed (changes requested):
Great work overall Corrie, just a few things to tidy up.

Aside from that lets increase the size of the nav bar

mrgiles1 commented:
Really like the NAVBAR Corrie. My initial thoughts were:
1. I'm a fan of the standard, right-side hamburger menu on mobile
2. I generally like my links, currently centered on desktop, to be either up against the logo (left) or up against the profile icon (right)
3. I mentioned the hamburger button not being vertically aligned with the links (seems a touch lower than them).
Other than those ideas, suggestions, love it.

Cililocwee committed:
separated components and add style.ts

Separated out the individual components in the navbar and start implementing an external style.ts file

Cililocwee committed:
Refactored mobile menu into Menu component

Cililocwee committed:
Changed font size for easier navigation

Minimized props on components and changed styles to be easier to navigate

Cililocwee commented:
I've changed how the mobile menu works/looks/aligns:

![image](https://github.com/jtabba/CDW-1/assets/12403004/211a816d-1861-4008-b0e1-441c3c83e83c)

The avatar/user profile icon is now tucked inside the menu:

![image](https://github.com/jtabba/CDW-1/assets/12403004/6cd9b968-374f-4286-b3ea-ad778311033e)
![image](https://github.com/jtabba/CDW-1/assets/12403004/5163a4d5-987a-46ed-b5da-9f2bfa8a026f)

The desktop menu is now bigger and easier to see:

![image](https://github.com/jtabba/CDW-1/assets/12403004/fad1b646-0cca-4253-bf2e-a99875ae859e)

Cililocwee committed:
Merge branch 'prod' into nav-bar

Cililocwee committed:
Fixed mobile menu font size

Cililocwee commented:
Brought prod to branch, ready for clean merge

jtabba commented on frontend/src/components/header/MobileNav.tsx:
```diff
@@ -0,0 +1,70 @@
+import { AuthButtons } from "../authentication/SignInOrOut";
+import { CloseIcon, HamburgerIcon } from "@chakra-ui/icons";
+
+const MobileNavItem = ({ label, children, href }: NavItem) => {
```
Keep it consistent and use FC

Cililocwee replied:
Got it! Finally, actually got it!

jtabba commented on frontend/src/components/header/MobileNav.tsx:
```diff
@@ -0,0 +1,70 @@
+  );
+};
+
+interface NavBarProps {}
```
Whats this for?

Cililocwee replied:
Style and pizzazz!!.....

It was a fundamental misunderstanding of your request for me to use FC :) I'll prune it

jtabba reviewed (changes requested):
Two small changes

Cililocwee committed:
Fixed props in MobileNav

Cililocwee commented:
Ready for reconsideration
